### PR TITLE
[SLA] PIM-7327: Fix completeness calculation memory leak

### PR DESCRIPTION
**Pull request by fitn**
Fix completeness calculation memory leak

Actual launch on 2048 products:
> romain@dandelion:~/workspace/pim/pim-ce$ time bin/console --env=prod pim:completeness:calculate
> Generating missing completenesses...
> 46.14
...
> 142.61
> Missing completenesses generated.
> real    2m41.547s

No real need to go further and try with 1M products, we can see there is a memory leak. Memory is increasing from 46M to 142M. There is no decrease (I don't put the different values but trust me :P).

**What is wrong with this code?!**
- First thing, we call the product saver. This introduces that the product will be saved again (Yes I am Captain Obvious). Saving a product has a cost, no matter if it has changed or not. Some events are dispatched and all subscribers and listeners will be executed. Doctrine will calculate a diff on the product to know if he has to save it or not (remember we use Deferred explicit change tracking policy): https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/change-tracking-policies.html 
- Second thing, we reindex our products. It's bad to reindex everything for a single modification but anyway, that's not the problem. Indexation is already called when we saved a product. The main concern of the product saver is to save somewhere and make the product available recoverable. A subscriber already exists in order to index: https://github.com/akeneo/pim-community-dev/blob/2.0/src/Pim/Bundle/CatalogBundle/EventSubscriber/IndexProductsSubscriber.php
Removing indexer calls reduces by 12.5% the execution time.
- Last thing and the one that makes me sad, the memory leak itself. I did a talk during the last PHP Tour explaining what happens in the UOW when you are working with high data volume. Clearing Doctrine cache ~~reduces the execution time from~~ prevent your script from being killed. Then, indeed, memory overload will reduces your execution time but it's not relevant on a such dataset. Here the memory increase is reduced by 2500% (from 46M to 50M).

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-2.0.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# 2.0.*
```
Comment (damien-carcel) [inline CHANGELOG-2.0.md]: 2.0.x, or Michel will not understand

File: src/Pim/Bundle/CatalogBundle/Command/CalculateCompletenessCommand.php [hunk 2]
```diff
@@ -62,6 +62,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
             if (count($productsToSave) === $container->getParameter('pim_job_product_batch_size')) {
                 $container->get('pim_catalog.saver.product')->saveAll($productsToSave);
                 $container->get('pim_catalog.elasticsearch.indexer.product')->indexAll($productsToSave);
+                $container->get('akeneo_storage_utils.doctrine.object_detacher')->detachAll($productsToSave);
```
Comment (damien-carcel) [inline src/Pim/Bundle/CatalogBundle/Command/CalculateCompletenessCommand.php]: Couldn't you use the CacheClearer instead of the detacher?

**Comment by damien-carcel** [comment]
Just a question. It is GTM for me whatever you decide.

File: src/Pim/Bundle/CatalogBundle/Command/CalculateCompletenessCommand.php (hunk 2 above)
Comment (fitn) [reply src/Pim/Bundle/CatalogBundle/Command/CalculateCompletenessCommand.php]: You're right, it will be faster! :)

Comment (BitOne) [reply src/Pim/Bundle/CatalogBundle/Command/CalculateCompletenessCommand.php]: Nope, the CacheClearer is something different than the objectDetacher. Here, you want to detach the product, not like clear the attribute or family cache ;)

Comment (juliensnz) [reply src/Pim/Bundle/CatalogBundle/Command/CalculateCompletenessCommand.php]: Or a simple and effective solution could be to do what we did on other commands lately: create a second command (private) and split the work in small chunks. The first command queries the products and pass an array of identifier to the second one. That way you don't fix the memory leak, you just don't care :)

Comment (fitn) [reply src/Pim/Bundle/CatalogBundle/Command/CalculateCompletenessCommand.php]: So I used cache clearer as seen with @BitOne 
@juliensnz This is not a solution. You call again the bootstrap. Clearing the entity manager (+ "cached" data in the proxy) is cleaner and resolves the memory leaks ;)

**Pull request merged**